File: .NET/Microsoft.Recognizers.Text.DateTime/DateTimeOptions.cs
using System;

namespace Microsoft.Recognizers.Text.DateTime
{
    [Flags]
    public enum DateTimeOptions
    {
        /// <summary>
        /// None
        /// </summary>
        None = 0,

        /// <summary>
        /// SkipFromToMerge
        /// </summary>
        SkipFromToMerge = 1,

        /// <summary>
        /// SplitDateAndTime
        /// </summary>
        SplitDateAndTime = 2,

        /// <summary>
        /// CalendarMode
        /// </summary>
        CalendarMode = 4,

        /// <summary>
        /// ExtendedTypes
        /// </summary>
        ExtendedTypes = 8,

        /// <summary>
        /// FailFast, mode that aborts extraction/tagging quickly for non-entity cases. May be removed later.
        /// </summary>
        FailFast = 2097152, // 2 ^21

        /// <summary>
        /// ExperimentalMode
        /// </summary>
        ExperimentalMode = 4194304, // 2 ^22

        /// <summary>
        /// EnablePreview
        /// </summary>
        EnablePreview = 8388608, // 2 ^23
    }
}

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/DateTimeOptions.java
package com.microsoft.recognizers.text.datetime;

public enum DateTimeOptions {
    None(0),
    SkipFromToMerge(1),
    SplitDateAndTime(2),
    CalendarMode(4),
    ExtendedTypes(8),
    EnablePreview(8388608),
    ExperimentalMode(4194304),
    ComplexCalendar(8 + 4 + 8388608);

    private final int value;

    DateTimeOptions(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public boolean match(DateTimeOptions option) {
        return (this.value & option.value) == option.value;
    }
}
